src/net_processing.cpp

```diff
@@ -2411,6 +2411,10 @@ bool ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRec
                       pfrom.nVersion.load(), pfrom.nStartingHeight,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToString()) : ""),
                       pfrom.m_tx_relay == nullptr ? "block-relay" : "full-relay");
+            if (!pfrom.m_tx_relay) {
+                std::vector<CAddress> anchors_to_save = connman->GetBlockRelayNodeAddresses();
+                DumpAnchors(connman->GetAnchorsDbPath(), anchors_to_save);
```

Comment:
Wait, why is this done on VERACK? Even if this was discussed in the PR, a comment would help.

Reply:
> Wait, why is this done on VERACK?

It is a point where we get know that an outbound connection to a block-relay-only peer has been established.

> Even if this was discussed in the PR, a comment would help.

What kind of comment do you expect to see here?